src/ProjectTemplates/Web.ProjectTemplates/content/Worker-FSharp/Worker.fs

```diff
@@ -0,0 +1,21 @@
+namespace Company.Application1
+
+open System
+open System.Collections.Generic
+open System.Linq
+open System.Threading
+open System.Threading.Tasks
+open Microsoft.Extensions.Hosting
+open Microsoft.Extensions.Logging
+
+type Worker(logger: ILogger<Worker>) =
+    inherit BackgroundService()
+
+    override _.ExecuteAsync(ct: CancellationToken) =
+        async {
+            while not ct.IsCancellationRequested do
+                logger.LogInformation("Worker running at: {time}", DateTimeOffset.Now)
+                do! Async.Sleep(1000)
```

Comment:
(caveat: I know nothing about FSharp)

Shouldn't you be passing the CancellationToken to Async.Sleep?

Reply:
Not necessary. Cancellation tokens automatically propagate with F# async. It's really neat.